File: app/src/main/java/dev/leonlatsch/kolibri/boot/jobs/base/AsyncJob.java
package dev.leonlatsch.kolibri.boot.jobs.base;

import android.content.Context;

/**
 * Base class for async jobs
 *
 * @author Leon Latsch
 * @since 1.0.0
 */
public abstract class AsyncJob extends BaseJob {

    private Thread thread;

    public AsyncJob(Context context) {
        super(context);
    }

    protected abstract void run(AsyncJobCallback asyncJobCallback);

    public void execute(AsyncJobCallback asyncJobCallback) {
        Runnable runnable = () -> run(asyncJobCallback);
        thread = new Thread(runnable, getThreadName());
        thread.start();
    }

    private String getThreadName() {
        return this.getClass().getName() + "-THREAD";
    }
}

File: app/src/main/java/dev/leonlatsch/kolibri/boot/jobs/base/AsyncJob.kt
package dev.leonlatsch.kolibri.boot.jobs.base

import android.content.Context

/**
 * Base class for async jobs
 *
 * @author Leon Latsch
 * @since 1.0.0
 */
abstract class AsyncJob(context: Context) : BaseJob(context) {

    private var thread: Thread? = null

    private val threadName: String
        get() = this.javaClass.name + "-THREAD"

    abstract fun execute(asyncJobCallback: AsyncJobCallback?)

    protected fun runAsync(runnable: Runnable) {
        thread = Thread(runnable, threadName)
        thread!!.start()
    }
}
